### DISCUSSION TITLE
convergence issue with AbauqsUMATstress

### DISCUSSION THREAD

**Discussion by edwardXZ06** [idaholab/moose#19574] [Q&A General] [answered]
Hi, everyone:

I met a question when running the AbaqusUMATstress feature in the MOOSE.

First the umat file works well with Abaqus regardless of element number. Second, there is no issue when running in the MOOSE on a single element test. However, when I just increase the element number, it becomes more and more difficult to converge. I have manipulated the executioner module a little bit but it doesn't work. 

Has anyone met similar issues or any advice on this problem? I have attached the input file in txt format. 

Many thanks!

Xiang
[elastic_test.txt](https://github.com/idaholab/moose/files/7671301/elastic_test.txt)

**Comment by ngrilli** [comment, marked as answer]
Dear @edwardXZ06 

Try using the adaptive time stepper, as explained here:
https://mooseframework.inl.gov/source/materials/abaqus/AbaqusUMATStress.html

It is because your UMAT may be asking a time step cut using PNEWDT,
but if you don't add the adaptive time stepper, the time step cut is not handled properly in moose.
Depending on your specific UMAT, that may cause problems.

You need to add the user objects, postprocessor and adaptive time stepper as explained here:
modules/tensor_mechanics/test/tests/umat/time_step/elastic_timestep.i

May not be the solution to your problem, but try anyway.

Best Regards,
Nicolò

**Comment by ngrilli** [comment]
@edwardXZ06 
Please check this other discussion, I think I ultimately found the origin of the convergence problem
when running the umat with more elements:
https://github.com/idaholab/moose/discussions/20109

Hope this helps
Nicolò
